tests/clickhouse/test_query_format.py

```diff
@@ -0,0 +1,196 @@
+                CurriedFunctionCall(
+                    "my_complex_math",
+                    FunctionCall(
+                        None,
+                        "doSomething",
+                        [
+                            Column(None, "column1", None),
+                            Column(None, "column2", "table1"),
+                            Column("al", "column3", None),
+                        ],
+                    ),
+                    [Column(None, "column1", None)],
+                )
+            ],
+            condition=binary_condition(
+                None,
+                "and",
+                lhs=binary_condition(
+                    None,
+                    "eq",
+                    lhs=Column("al", "column3", None),
+                    rhs=Literal(None, "blabla"),
+                ),
+                rhs=binary_condition(
+                    None,
+                    "neq",  # yes, not very smart
+                    lhs=Column("al", "column3", None),
+                    rhs=Literal(None, "blabla"),
+                ),
+            ),
+            groupby=[
+                CurriedFunctionCall(
+                    "my_complex_math",
+                    FunctionCall(
+                        None,
+                        "doSomething",
+                        [
+                            Column(None, "column1", None),
+                            Column(None, "column2", "table1"),
+                            Column("al", "column3", None),
+                        ],
+                    ),
+                    [Column(None, "column1", None)],
+                )
+            ],
+            order_by=[
+                OrderBy(
+                    OrderByDirection.ASC,
+                    FunctionCall(None, "f", [Column(None, "column1", None)]),
+                )
+            ],
+        ),
+        (
+            "SELECT (doSomething(column1, table1.column2, (column3 AS al))(column1) AS my_complex_math) "
+            "FROM my_table "
+            "WHERE and(eq(al, 'blabla'), neq(al, 'blabla')) "
+            "GROUP BY (my_complex_math) "
+            "ORDER BY f(column1) ASC"
+        ),
+    ),
+    (
+        # Query with escaping
+        Query(
+            {},
+            TableSource("my_table", ColumnSet([])),
+            selected_columns=[
+                Column("al1", "field_##$$%", None),
+                Column("al2", "f@!@", "t&^%$"),
+            ],
+            groupby=[
+                Column("al1", "field_##$$%", None),
+                Column("al2", "f@!@", "t&^%$"),
+            ],
+            order_by=[OrderBy(OrderByDirection.ASC, Column(None, "column1", None))],
+        ),
+        (
+            "SELECT (`field_##$$%` AS al1), (`t&^%$`.`f@!@` AS al2) "
+            "FROM my_table "
+            "GROUP BY (al1, al2) "
+            "ORDER BY column1 ASC"
+        ),
+    ),
+]
+
+
+@pytest.mark.parametrize("query, formatted", test_cases)
```

Comment:
Not really in scope for this change, but it seems like we're following this pattern of query test parameterization in several places — would it make sense later to just define a set of fixtures queries that are closer to a comprehensive test of the different clauses and expression types? Or would generalizing the fixtures to that degree do more harm than good?

Reply:
So far the expressions/queries I used for testing expression, formatting, and parsing where quite different from each other since I was stressing different areas. I think we could do that for parsing though.